### PR TITLE
Bug Fix - Scanning circular dependencies

### LINKED ISSUE #76: [BUG] dependency with circular dependencies causes StackOverflowError
**Describe the bug**
I you have a dependency which has circular dependencies (e.g. 'org.apache.xmlgraphics:batik-transcoder:1.7') you receive a java.lang.StackOverflowError when running ossIndexAudit

**To Reproduce**
build.gradle 
```
plugins {
    id 'java'
    id 'org.sonatype.gradle.plugins.scan' version '2.0.8' 
}
repositories {
   mavenCentral()
}
dependencies {
    implementation 'org.apache.xmlgraphics:batik-transcoder:1.7'
}
```
Steps to reproduce the behavior
1. Run `./gradlew ossIndexAudit` with attached 
2. Get error
```
> Task :ossIndexAudit FAILED
Using anonymous request
POM relocation to an other version number is not fully supported in Gradle : xml-apis:xml-apis:2.0.2 relocated to xml-apis:xml-apis:1.0.b2.
Please update your dependency to directly use the correct version 'xml-apis:xml-apis:1.0.b2'.
Resolution will only pick dependencies of the relocated element.  Artifacts and other metadata will be ignored.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ossIndexAudit'.
> java.lang.StackOverflowError (no error message)
```
**Expected behavior**
Expect to get output from ossIndexAudit

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: window 10
 - Gradle Version: 6.8.3
 - JVM Version and Flavor: Java 8
 - Nexus IQ Server Version: None
 - Version 2.0.8

**Additional context**
Moving up to batik 1.8 does work (I'm assuming the circular dependencies were resolved.) Not sure what other project may have this issue.

### PR DESCRIPTION
**Pull request by shaikhu**
Tested using a dependency (`org.apache.xmlgraphics:batik-transcoder:1.7`) which contains circular dependencies. Was able to produce report. Will leave in draft for further testing.

It relates to the following issue #s:
* Fixes #76 

cc @bhamail / @DarthHater / @guillermo-varela / @shaikhu

### CODE REVIEW

**Comment by guillermo-varela** [comment]
Tested for IQ and the stackoverflow error is also present but in a difference place:

```
Caused by: java.lang.StackOverflowError
        at com.google.common.collect.ImmutableSet.copyOf(ImmutableSet.java:251)
        at org.gradle.api.internal.artifacts.DefaultResolvedDependency.getChildren(DefaultResolvedDependency.java:97)
        at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.processDependency(DependenciesFinder.java:188)
        at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$processDependency$12(DependenciesFinder.java:188)
        at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.processDependency(DependenciesFinder.java:188)
```

**Comment by guillermo-varela** [comment]
A full list of dependencies is not being shown for IQ scans.
Further investigation is needed to check if the issue lies on `nexus-platform-api`.

**Comment by guillermo-varela** [comment]
An issue was found in `nexus-platform-api`, a PR was raised to address it: https://github.com/sonatype/nexus-java-api/pull/162

File: build.gradle [hunk 1]
```diff
@@ -78,7 +78,8 @@ dependencies {
   shadow localGroovy()
   shadow gradleApi()
 
-  implementation files ("nexus-platform-api/build/libs/nexus-platform-api-${version}.jar")
+  //implementation files ("nexus-platform-api/build/libs/nexus-platform-api-${version}.jar")
```
Comment (guillermo-varela) [inline build.gradle:81]: I believe this is needed just locally to workaround IDE issues.

Comment (shaikhu) [reply build.gradle]: > I believe this is needed just locally to workaround IDE issues.

:sweat_smile: beab896d

**Comment by guillermo-varela** [comment]
` nexus-platform-api` has been updated and the dependency graph sent to NexusIQ was improved to avoid circular references there.

**Comment by shaikhu** [comment]
> ` nexus-platform-api` has been updated and the dependency graph sent to NexusIQ was improved to avoid circular references there.

LGTM :+1:

**Review by guillermo-varela — APPROVED** [review]
+1

**Pull request merged**